services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImplV2.java

```diff
@@ -319,7 +429,161 @@ private Long saveNewsIllustration(String uploadId, Long oldFileId) {
+    if (draftPage == null) {
+      throw new EntityNotFoundException("draft page not found");
+    }
+    Page rootPage = getArticlesRootPage(draftPage.getWikiOwner());
+    if (rootPage == null) {
+      throw new EntityNotFoundException("Articles root page not found");
+    }
+    DraftPage draftArticle = new DraftPage();
+    draftArticle.setId(news.getId());
+    draftArticle.setTargetPageId(null);
+    draftArticle.setTitle(news.getTitle());
+    draftArticle.setContent(news.getBody());
+    draftArticle.setParentPageId(rootPage.getId());
+    draftArticle.setAuthor(news.getAuthor());
+    // created and updated date set by default during the draft creation
+    // process
+    draftArticle = noteService.updateDraftForNewPage(draftArticle, System.currentTimeMillis());
+    if (draftArticle != null) {
+      NewsDraftObject newsDraftMetaDataObject = new NewsDraftObject(NEWS_METADATA_DRAFT_OBJECT_TYPE, draftArticle.getId(), null);
+      MetadataKey newsDraftMetadataKey = new MetadataKey(NEWS_METADATA_TYPE.getName(), NEWS_METADATA_NAME, 0);
+      List<MetadataItem> metadataItems = metadataService.getMetadataItemsByMetadataAndObject(newsDraftMetadataKey, newsDraftMetaDataObject);
+      if (metadataItems != null && !metadataItems.isEmpty()) {
+        MetadataItem existingMetadata = metadataItems.get(0);
+        Map<String, String> properties = existingMetadata.getProperties();
+        if (properties == null) {
+          properties = new HashMap<>();
+        }
+        // create or update the illustration
+        if (news.getUploadId() != null && !news.getUploadId().isBlank()) {
+          if (properties.containsKey(NEWS_UPLOAD_ID) && properties.get(NEWS_UPLOAD_ID) != null && !properties.get(NEWS_UPLOAD_ID).equals(news.getUploadId())) {
+            FileItem fileItem = fileService.getFile(Long.parseLong(properties.get(NEWS_ILLUSTRATION_ID)));
+            Long illustrationId = saveNewsIllustration(news.getUploadId(), fileItem.getFileInfo().getId());
+            properties.put(NEWS_ILLUSTRATION_ID, String.valueOf(illustrationId));
+            setNewsIllustration(news, illustrationId, DRAFT.name());
+          } else {
+            Long illustrationId = saveNewsIllustration(news.getUploadId(), null);
+            properties.put(NEWS_ILLUSTRATION_ID, String.valueOf(illustrationId));
+            setNewsIllustration(news, illustrationId, DRAFT.name());
+          }
+          properties.put(NEWS_UPLOAD_ID, news.getUploadId());
+        }
+        if (news.getSummary() != null && !news.getSummary().isBlank()) {
+          properties.put(NEWS_SUMMARY, news.getSummary());
+        }
+        existingMetadata.setProperties(properties);
+        String identityId = identityManager.getOrCreateUserIdentity(updater).getId();
+        metadataService.updateMetadataItem(existingMetadata, Long.parseLong(identityId));
+      }
+    }
+    return news;
+  }
+
+  private boolean canEditNews(News news, String authenticatedUser) {
```

Comment:
this method is updated in NewsServiceImpl to be updated here

Reply:
method updated !